R/class-Collections.R

```diff
@@ -0,0 +1,415 @@
+isNAorNonNegative <- function(x) {
+    return(is.na(x) | x >= 0)
+}
+
+check_collection <- function(object) {
+    errors <- character()
+    df <- object@data
+    allIdColumns <- c(object@recordIdColumn, object@ancestorIdColumns)
+
+    # check that name is not empty
+    if (object@name == "") {
+        msg <- "name cannot be empty"
+        errors <- c(errors, msg)
+    }
+
+    # Ensure id columns are valid
+    msg <- validateIdColumns(df, object@recordIdColumn, object@ancestorIdColumns)
+    errors <- c(errors, msg)
+
+    # check that all columns in data are numeric except id columns
```

Comment:
if/when categorical collections come around, do you see that as a new subclass or it all handled in this class?

Reply:
i havent gotten that far..

Reply:
lol i know and i dont expect anything like that to happen for a long time, but just curious if you had thoughts yet

Reply:
i suspect itd be a refactor to the existing Collection, CollectionWithMetadata classes, and the addition of new subclasses. but idk